odoo/addons/test_lint/tests/test_pofile.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import Counter
+
+from odoo.modules import get_modules, get_resource_path
+from odoo.tests.common import TransactionCase
+from odoo.tools.translate import TranslationFileReader
+
+
+class PotLinter(TransactionCase):
+    def test_pot_files(self):
+        def format_key(entry):
+            if entry['type'] == 'model':
+                return ('model', entry['name'], entry['imd_name'])
+            elif entry['type'] == 'model_terms':
+                return ('model_terms', entry['name'], entry['imd_name'], entry['src'])
+            elif entry['type'] == 'code':
+                return ('code', entry['src'])
+
+        # retrieve all modules, and their corresponding POT file
+        for module in get_modules():
+            filename = get_resource_path(module, 'i18n', module + '.pot')
+            if not filename:
+                continue
+            counts = Counter(
+                format_key(entry)
+                for entry in TranslationFileReader(filename)
+            )
+            duplicates = [key for key, count in counts.items() if count > 1]
```

Comment:
We may want to `counts.pop(None)` beforehand, as there seem to be pot entries with other types than those.

Though I'm not sure those types make any sense or have any relevance: there are entries of type `view` (?) and `field` (????) in the `google_base_account` module, as well as entries of type `selection` in pos_blackbox_be and l10n_co_edi.

Reply:
And since the key extraction was, well, extracted, to a function, that seems like a good use case for `map`.

Reply:
There are indeed other entry types in `pot` files, but `TranslationFileReader` only returns those three types (and ignores the other ones).

Reply:
Maybe we should add a warning if the entry type is unknown, so we remember to update this lint for the new type in case we decided to add a new one in the future?